matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java

```diff
@@ -259,20 +261,18 @@ public boolean isLocal()
     public Collection<String> getEmails()
     {
         Collection<String> emails = new LinkedList<>();
-        if (this.isLocal()) {
-            // Note: The patient is not saved because sometimes (often?) this method is run not on an object that
-            // was just created, but on one created from a column from the DB. In this case, we might not have
-            // a Patient object when the patient is remote.
-            emails.addAll(this.getPatientEmails(this.patient));
-        } else {
-            if (StringUtils.isNotEmpty(this.href)) {
+        if (this.contactInfo != null) {
+            if (this.isLocal()) {
+                emails.addAll(this.contactInfo.getEmails());
+            } else {
                 // TODO: discuss what better algorithm/package can be use to split emails
-                // MME emails may be of the form "mailto:email1,email2", need ot parse that
-                if (this.href.startsWith("mailto:")) {
-                    String emailList = this.href.replace("mailto:", "");
+                // MME emails may be of the form "mailto:email1,email2", need to parse that
+                String href = this.contactInfo.getUrl();
+                if (href.startsWith("mailto:")) {
+                    String emailList = href.replace("mailto:", "");
                     emails.addAll(Arrays.asList(emailList.split(",|;")));
                 } else {
-                    emails.add(this.href);
```

Comment:
this is the line which used that piece of data (...and which used to be transferred to front-end as one of the emails)